contracts/psp22/lib.rs

```diff
@@ -0,0 +1,153 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+#[ink::contract]
+mod psp22 {
+
+    use ink::{codegen::EmitEvent, reflect::ContractEventBase, storage::Mapping};
+    use psp22_traits::{PSP22Error, PSP22};
+
+    #[ink(event)]
+    pub struct Approval {
+        #[ink(topic)]
+        owner: AccountId,
+        #[ink(topic)]
+        spender: AccountId,
+        value: Balance,
+    }
+
+    #[ink(event)]
+    pub struct Transfer {
+        #[ink(topic)]
+        from: Option<AccountId>,
+        #[ink(topic)]
+        to: Option<AccountId>,
```

Comment:
we don't need `Option`, do we?

Reply:
indeed, initially i emitted an event in constructor but thats no longer the case - removed